src/control/server/init/setup_spdk.sh

```diff
@@ -5,11 +5,67 @@
 ## to be run by an unprivileged user).
 ## These sudo commands can be granted using visudo by a system administrator.
 
-set -e
+set +e
+
+#
+# Sets appropriate permissions on /dev/vfio/* files
+#
+set_vfio_permissions()
+{
+	# make sure regular users can read /dev/vfio
+	echo "RUN: chmod /dev/vfio"
+	sudo chmod a+x /dev/vfio
+	if [ $? -ne 0 ] ; then
+		echo "FAIL"
+		return
+	fi
+	echo "OK"
+
+	# make sure regular user can access everything inside /dev/vfio
+	echo "RUN: chmod /dev/vfio/*"
+	sudo chmod 0666 /dev/vfio/*
+	if [ $? -ne 0 ] ; then
+		echo "FAIL"
+		return
+	fi
+	echo "OK"
+
+	# since permissions are only to be set when running as
+	# regular user, we only check ulimit here
+	#
+	# warn if regular user is only allowed
+	# to memlock <64M of memory
+	MEMLOCK_AMNT="$(ulimit -l)"
+
+	if [ "$MEMLOCK_AMNT" != "unlimited" ] ; then
+		MEMLOCK_MB="$(($MEMLOCK_AMNT / 1024))"
+		echo ""
+		echo "Current user memlock limit: ${MEMLOCK_MB} MB"
+		echo ""
+		echo "This is the maximum amount of memory you will be"
+		echo "able to use with DPDK and VFIO if run as current user."
+		echo -n "To change this, please adjust limits.conf memlock "
+		echo "limit for current user."
+
+		if [ "$MEMLOCK_AMNT" -lt 65536 ] ; then
+			echo ""
+			echo "## WARNING: memlock limit is less than 64MB"
+			echo -n "## DPDK with VFIO may not be able to "
+			echo "initialize if run as current user."
+		fi
+	fi
```

Comment:
We're doing other memory-related checks in server.go ... I wonder it if would make more sense to pull this out of the script and move it there.